src/explorer/Components/DistributionAnalysisComponent.cs

```diff
@@ -0,0 +1,100 @@
+namespace Explorer.Components
+{
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading.Tasks;
+
+    using Accord.Statistics.Analysis;
+    using Accord.Statistics.Distributions.Univariate;
+    using Accord.Statistics.Testing;
+    using Explorer.Metrics;
+
+    public class DistributionAnalysisComponent : ExplorerComponent<GoodnessOfFitCollection>, PublisherComponent
+    {
+        private readonly ResultProvider<EmpiricalDistribution> distributionProvider;
+
+        public DistributionAnalysisComponent(ResultProvider<EmpiricalDistribution> distributionProvider)
+        {
+            this.distributionProvider = distributionProvider;
+        }
+
+        public async IAsyncEnumerable<ExploreMetric> YieldMetrics()
+        {
+            var fits = await ResultAsync;
+
+            var goodFits = fits.Values
+                .Where(fit =>
+                    fit.AndersonDarlingRank == 0 ||
+                    fit.ChiSquareRank == 0 ||
+                    fit.KolmogorovSmirnovRank == 0);
+
+            yield return new UntypedMetric(
+                name: "distribution_estimates",
+                metric: goodFits.Select(fit =>
+                {
+                    var ad = fit.Analysis.AndersonDarling[fit.Index];
+                    var cs = fit.Analysis.ChiSquare[fit.Index];
+                    var ks = fit.Analysis.KolmogorovSmirnov[fit.Index];
+
+                    return new
+                    {
+                        fit.Name,
+                        Distribution = fit.Distribution.ToString(),
+                        Goodness = new GoodnessMetric?[]
+                        {
+                            ad is null
```

Comment:
It is not clear why you add null elements in the metrics here.

Reply:
I couldn't think of any way to 'add nothing' so I added `null` and then filter out the null values in the next step (line 55).